src/targets/shell/curl/client.ts

```diff
@@ -0,0 +1,134 @@
+/**
+ * @description
+ * HTTP code snippet generator for the Shell using cURL.
+ *
+ * @author
+ * @AhmadNassri
+ *
+ * for any questions or issues regarding the generated code snippet, please open an issue mentioning the author.
+ */
+
+import { CodeBuilder } from '../../../helpers/code-builder';
+import { getHeaderName } from '../../../helpers/headers';
+import { quote } from '../../../helpers/shell';
+import { Client } from '../../targets';
+
+export interface CurlOptions {
+  short?: boolean;
+  binary?: boolean;
+  globOff?: boolean;
+}
+
+export const curl: Client<CurlOptions> = {
+  info: {
+    key: 'curl',
+    title: 'cURL',
+    link: 'http://curl.haxx.se/',
+    description: 'cURL is a command line tool and library for transferring data with URL syntax',
+  },
+  convert: ({ fullUrl, method, httpVersion, headersObj, allHeaders, postData }, options) => {
+    const opts = {
+      indent: '  ',
+      short: false,
+      binary: false,
+      globOff: false,
+      ...options,
+    };
+    const { push, join } = new CodeBuilder({
+      indent: opts.indent,
+      // @ts-expect-error SEEMS LEGIT
+      join: opts.indent !== false ? ` \\\n${opts.indent}` : ' ',
+    });
+
+    const globOption = opts.short ? '-g' : '--globoff';
+    const requestOption = opts.short ? '-X' : '--request';
+    let formattedUrl = quote(fullUrl);
+
+    push(`curl ${requestOption} ${method}`);
+    if (opts.globOff) {
+      formattedUrl = unescape(formattedUrl);
+      push(globOption);
+    }
+    push(`${opts.short ? '' : '--url '}${formattedUrl}`);
+
+    if (httpVersion === 'HTTP/1.0') {
+      push(opts.short ? '-0' : '--http1.0');
+    }
+
+    // if multipart form data, we want to remove the boundary
+    if (postData.mimeType === 'multipart/form-data') {
+      const contentTypeHeaderName = getHeaderName(headersObj, 'content-type');
+      if (contentTypeHeaderName) {
+        const contentTypeHeader = headersObj[contentTypeHeaderName];
+        if (contentTypeHeaderName && contentTypeHeader) {
+          // remove the leading semi colon and boundary
+          // up to the next semi colon or the end of string
+          // @ts-expect-error SEEMS LEGIT
+          const noBoundary = contentTypeHeader.replace(/; boundary.+?(?=(;|$))/, '');
```

Comment:
Also weird that `contentTypeHeader` might be a `string[]` sometimes.

Reply:
oh, for sure it's strange.. but this is how the app worked all along!

see https://github.com/Kong/httpsnippet/blob/master/src/helpers/reducer.js#L15 which is used to build up the `headersObj` object.  The values can be string arrays!!  there were even tests for this behavior https://github.com/Kong/httpsnippet/blob/master/test/reducer.js#L22.  This is a bug, in my opinion, that PRETTY MUCH all of the clients are written as though this cannot happen, when in fact it definitely can.  It's an edge case, for sure, but a real one all the same.

<img src="https://media4.giphy.com/media/l0IsIsuNJly1evlIY/giphy.gif"/>

Reply:
I added some documentation on this in 2ffb3da4a84b48cc018acc5446bf6197e0253963